addons/udes_stock/models/stock_picking.py

```diff
@@ -3093,3 +3098,43 @@ def open_back_orders(self):
         return self.open_related_pickings(
             self.u_created_back_orders.ids, "Created Back Orders"
         )
+
+    @api.multi
+    def safe_prevent_date_done_on_not_done_pickings(self):
+        """
+        Set date_done field to empty in case that not all moves are in state done
+        """
+        for picking, moves in self.mapped("move_lines").groupby("picking_id"):
+            if picking.date_done and moves and not all(
+                    move.state in ("done", "cancel") for move in moves
```

Comment:
does this mean, that if all move lines are cancelled - the picking won't have its date done cleared?

we might need a secondary check - `if all(move.state == "cancel" for move in moves)` then date_done should be cleared